core/metacling/src/TClingClassInfo.h

```diff
@@ -89,7 +89,11 @@ class TClingClassInfo {
    void                 DeleteArray(void *arena, bool dtorOnly, const ROOT::TMetaUtils::TNormalizedCtxt &normCtxt) const;
    void                 Destruct(void *arena, const ROOT::TMetaUtils::TNormalizedCtxt &normCtxt) const;
    const clang::ValueDecl *GetDataMember(const char *name) const;
-   const clang::Decl      *GetDecl() const { return fDecl; } // Underlying representation without Double32_t
+   void SetDecl(const clang::Decl* D) {
+     // FIXME: We should track down all sets and potentially avoid them.
+     fDecl = D;
+     fNameCache = {}; // invalidate the cache.
```

Comment:
Is that 'better' than ```fNameCache.clear();```

Reply:
Clear should be faster as it does not make a copy I think. Quickbench shows it is 5 times faster on some implementations...

Reply:
Cool.  I also find it semantically clearer :)